app/webpacker/components/Panel/views/CronJobStatus/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {
+  Button, Header, List, Message,
+} from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
+import { DateTime } from 'luxon';
+import _ from 'lodash';
+import I18n from '../../../../lib/i18n';
+import getCronjobDetails from './api/getCronjobDetails';
+import Loading from '../../../Requests/Loading';
+import Errored from '../../../Requests/Errored';
+import CronjobActions from './CronjobActions';
+
+const CRONJOB_MESSAGE_TYPES = ['warning', 'info', 'error', 'success'];
+const CRONJOB_MESSAGE_TYPES_MAP = _.keyBy(CRONJOB_MESSAGE_TYPES, (item) => item);
```

Comment:
I think this is taking type safety a bit too far. We already know we're definitely going to use TypeScript in NextJS, at which point this kind of "type-safe map accessing" will become completely redundant.

Reply:
If you feel very strongly about it, then I won't block the whole PR because of this.
But it just feels like you're creating too much work for such a simple purpose.

Reply:
Yes, I think that will be fine.

Reply:
For future reference, please note that `CRONJOB_MESSAGE_TYPES_MAP` is only used within this file. If...
- it holds shared constants that are `export`ed
- or it reads data from the backend (ie currently our `wca-data.js-erb` hack)

then it makes sense to "secure" the data access like that. So your approach is not "forbidden" in general.

But in this particular case, you're using four hand-written literal string constants and then wrapping them in an array with key access. This is not much better than just writing the string literals anyways.

Reply:
Understood, changed it.